phpunit/html/wp-html-walker-test.php

```diff
@@ -0,0 +1,469 @@
+<?php
+/**
+ * Unit tests covering WP_HTML_Walker functionality.
+ *
+ * @package WordPress
+ * @subpackage HTML
+ */
+
+/**
+ * @group html
+ */
+class Tests_HTML_WP_HTML_Walker extends WP_UnitTestCase {
```

Comment:
The tests would need changes when the `WP_HTML_Walker` and related classes are encapsulated so they are not accessible outside of the block parser.